flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobGraph.java

```diff
@@ -352,8 +353,8 @@ public JobVertex findVertexByID(JobVertexID id) {
 	 *
 	 * @param paths paths of the directories/JAR files required to run the job on a task manager
 	 */
-	public void setClasspaths(List<URL> paths) {
-		classpaths = paths;
+	public void addClasspaths(Collection<URL> paths) {
+		classpaths.addAll(paths);
 	}
 
 	public List<URL> getClasspaths() {
```

Comment:
Maybe we shouldn't allow modifications of this list from outside.

Reply:
Make sense to return an unmodifiableList